### PR TITLE
some fixes and cleaning

### PR DESCRIPTION
**Pull request by brianmcgillion**
## Description of changes

Cleanup and minor bug fixes

## Checklist for things done

- [X] Summary of the proposed changes in the PR description
- [X] More detailed description in the commit message(s)
- [X] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [X] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [X] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status

### CODE REVIEW

File: targets/lenovo-x1/guivmExtraModules.nix [hunk 1]
```diff
@@ -150,10 +153,15 @@
...
     };
 
-    # Open TCP port for the PDF XDG socket
-    networking.firewall.allowedTCPPorts = [xdgPdfPort];
+    # Enable all firmware for graphics firmware
+    hardware.enableRedistributableFirmware = true;
+    hardware.enableAllFirmware = true;
     # Early KMS needed for GNOME to work inside GuiVM
     boot.initrd.kernelModules = ["i915"];
+    boot.kernelParams = ["i915.force_probe=a7a1" "earlykms"];
```
Comment (mbssrc) [inline targets/lenovo-x1/guivmExtraModules.nix:161]: `"i915.force_probe=a7a1"` - this is now hardcoded for **x1 gen11** only, probably best to get the current passthrough gpu device. Also, does this achieve anything?

**Comment by leivos-unikie** [comment]
Tested on Lenovo-X1 (boot from USB SSD)
- All apps launch
- ci-test-automation tests pass but there are some deviations in performance (I will paste results after I have got reference from current mainline)

Boot time to ghaf desktop felt longer than before. So I measured it (including 5s counter at NixOS menu). 
- This PR: 1 min 14s
- ghaf mainline build #166 image: 44s

The screen is black quite a long time after boot logs are gone. Could the display show somethign like "Starting Ghaf" (if it is necessary to have such a long boot ime after all).

Shutdown (ghaf desktop poweroff button) on the other hand is very fast, less than 5s.

**Comment by brianmcgillion** [comment]
> Tested on Lenovo-X1 (boot from USB SSD)
> 
> * All apps launch
> * ci-test-automation tests pass but there are some deviations in performance (I will paste results after I have got reference from current mainline)
> 
> Boot time to ghaf desktop felt longer than before. So I measured it (including 5s counter at NixOS menu).
> 
> * This PR: 1 min 14s
> * ghaf mainline build 166 image: 44s
> 
> The screen is black quite a long time after boot logs are gone. Could the display show somethign like "Starting Ghaf" (if it is necessary to have such a long boot ime after all).
> 
> Shutdown (ghaf desktop poweroff button) on the other hand is very fast, less than 5s.

Interesting I measured 47 seconds yesterday on my laptop, using a watch timer. I had the same feeling of slow. I will run it again to see

**Comment by leivos-unikie** [comment]
The performance issue didn't come from this PR. Those performance deviations came already at previous merge (build 166) to mainline.

**Review by mbssrc — APPROVED** [review]

**Pull request merged**